src/NuGet.Services.Metadata.Catalog.Monitoring/ValidationCollectorInitializer.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using Microsoft.Extensions.Logging;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Services.Metadata.Catalog.Persistence;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Assists with initializing a <see cref="ValidationCollector"/>.
+    /// </summary>
+    public class ValidationCollectorInitializer
+    {
+        private ILoggerFactory _loggerFactory;
+        private ILogger<ValidationCollectorInitializer> _logger;
+
+        public class Context
```

Comment:
made read-only and renamed it

it's a nested type because
1 - I didn't want to pass around a `Tuple`
2 - it only really makes sense within the context of the `ValidationCollectorFactory`